homeassistant/components/velux/__init__.py

```diff
@@ -46,13 +47,29 @@ class VeluxModule:
 
     def __init__(self, hass, config):
         """Initialize for velux component."""
-        from pyvlx import PyVLX
+        self._hass = hass
+        self._config = config
+        
 
-        host = config[DOMAIN].get(CONF_HOST)
-        password = config[DOMAIN].get(CONF_PASSWORD)
+    def setup(self):
+        async def on_hass_stop(event):
+            """Close connection when hass stops."""
+            _LOGGER.info("Velux interface terminated")
+            await self.pyvlx.disconnect()
+            
+        self._hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, on_hass_stop)     
+        from pyvlx import PyVLX
+        host = self._config[DOMAIN].get(CONF_HOST)
+        password = self._config[DOMAIN].get(CONF_PASSWORD)
         self.pyvlx = PyVLX(host=host, password=password)
 
+
     async def async_start(self):
         """Start velux component."""
+        _LOGGER.info("Velux interface started")        
         await self.pyvlx.load_scenes()
         await self.pyvlx.load_nodes()
+                        
+        for node in self.pyvlx.nodes:
+          _LOGGER.info("Stopping node: " + str(node))
+          await node.stop(wait_for_completion=False)
```

Comment:
Why do we need to stop the nodes here?

Reply:
I'm sorry.. It seemed to sneak its way in there as its part of another fix that does not belong to this particular PR.

should be removed in latest commit.